### PR TITLE
Add a test for connection handler

### PR DESCRIPTION
**Pull request by Nek-**
It also adds a new warning message in logs in case you're not using the right handler.

Finally it authorize new dependencies version for nekland/tools.

### CODE REVIEW

**Comment by Nek-** [comment]
Test suite all green with wstest. @Awkan can you merge it please ? 😇

File: src/Server/WebSocketServer.php [hunk 1]
```diff
@@ -188,9 +188,11 @@ private function getMessageHandler(string $uri, Connection $connection)
                 $handler = new $handler;
             }
 
+            $handler->onConnection($connection);
```
Comment (mcsky) [inline src/Server/WebSocketServer.php]: 👏 😂

File: src/Server/WebSocketServer.php [hunk 2]
```diff
@@ -188,9 +188,11 @@ private function getMessageHandler(string $uri, Connection $connection)
...
             return $handler;
         }
 
+        $this->logger->warning('Connection on ' . $uri . ' but no handler found.');
```
Comment (mcsky) [inline src/Server/WebSocketServer.php]: 👍

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 
 ## [Unreleased]
+### Changed
+- Allow nekland/tools in 2.0 version (sill works with 1.0)
```
Comment (Awkan) [inline CHANGELOG.md]: still

**Comment by Nek-** [comment]
Thanks @Awkan :) .

**Pull request merged**